src/crush/mapper.c

```diff
@@ -304,6 +314,15 @@ static int bucket_straw2_choose(struct crush_bucket_straw2 *bucket,
 
 	for (i = 0; i < bucket->h.size; i++) {
 		w = bucket->item_weights[i];
+		if (reweight_items && work->picked_sum > 0) {
+			__u64 v = w;
+			v *= (work->weight_sum -
+			      work->picked_sum);
+			v /= work->weight_sum;
```

Comment:
Define
```
#define div64_u64(dividend, divisor) ((dividend) / (divisor))
```
macro in crush_compat.h and use it here (see div64_s64() usage below).